src/main/java/juja/microservices/gamification/service/AchievementService.java

```diff
@@ -70,4 +73,40 @@ public String addDaily(String userFromId, String description) {
         }
         return result;
     }
+
+    public List<String> addCodenjoy(String  userFromId, String firstUserToId, String secondUserToId,
+            String thirdUserToId) {
+        List<String> result = new ArrayList<>();
+        List<Achievement> codenjoyUsersToday = achievementRepository.getAllCodenjoyAchievementsCurrentDate();
+        if (!codenjoyUsersToday.isEmpty()) {
+            throw new UnsupportedAchievementException("You cannot give codenjoy points twice a day");
+        } else if ("".equals(userFromId)) {
+            throw new UnsupportedAchievementException("User from cannot be empty");
+        } else if (!"".equals(secondUserToId) && "".equals(firstUserToId)) {
+            throw new UnsupportedAchievementException("First user cannot be empty");
+        } else if (!"".equals(thirdUserToId) &&  "".equals(secondUserToId)) {
+            throw new UnsupportedAchievementException("Second user cannot be empty");
+        } else if (!"".equals(secondUserToId) && firstUserToId.equalsIgnoreCase(secondUserToId)) {
+            throw new UnsupportedAchievementException("First and second users must be different");
+        } else if (!"".equals(thirdUserToId) && firstUserToId.equalsIgnoreCase(thirdUserToId)) {
+            throw new UnsupportedAchievementException("First and third users must be different");
+        } else if (!"".equals(secondUserToId) && !"".equals(thirdUserToId)
+                && secondUserToId.equalsIgnoreCase(thirdUserToId)) {
+            throw new UnsupportedAchievementException("Second and third users must be different");
+        }
+        Achievement firstThanks = new Achievement(userFromId, firstUserToId, CODENJOY_FIRST_PLACE,
```

Comment:
I think the correct name of achievement is firstPlace, secondPlace, thirdPlace. But not thanks.